fix processing of shipping method

tuunit opened the pull request:
Several Magento extensions use suffixes for the shipment method labels to
differentiate between the different DHL package types. For this reason no
'Create Label' checkbox is displayed in the backend for these extensions.
Because the onProcessMethods function only returns true if the label exactly
matches the extensions carrier name.

One example extension would be the 'MatrixRate Table Rate Shipping' by
WebShopApps. Which generate label names like 'matrixrate_500' for 'DHL Paket'
and 'matrixrate_509' for 'DHL Paket International', but does have the carrier
name 'matrixrate'.

tuunit committed:
fix processing of shipping method
Several Magento extensions use suffixes for the shipment method labels to
differentiate between the different DHL package types. For this reason no
'Create Label' checkbox is displayed in the backend for these extensions.
Because the onProcessMethods function only returned true if the label exactly
matches the extensions carrier name.

One example extension would be the 'MatrixRate Table Rate Shipping' by
WebShopApps. Which generate label names like 'matrixrate_500' for 'DHL Paket'
and 'matrixrate_509' for 'DHL Paket International', but does have the carrier
name 'matrixrate'.

powli commented on Model/Config/ModuleConfig.php:
```diff
@@ -189,7 +189,14 @@ public function canProcessRoute($destinationCountryId, $store = null)
-        return in_array($shippingMethod, $this->getShippingMethods($store));
+        $configuredMethods = $this->getShippingMethods($store);
+        foreach ($configuredMethods as $method) {
+            if (strpos($shippingMethod, $method)) {
```
`strpos($shippingMethod, $method)` can return 0 as a valid result, which will result in a false negative. It's better to do a type save check here `strpos($shippingMethod, $method) !== false`.

tuunit committed:
fix processing of shipping method
add type save check for because return value 0 would result in a false negative

powli commented:
@tuunit: Change has been merged in our internal repo and will be released in this repository with the next public release. Thanks for your contribution.

MarcusWolschon commented:
Why do you close issues in this repository when they have not yet been fixed in THIS repository?
How are people downloading what they currently find here know about contained, known bugs ?
When is "next public release"?

ngolatka commented:
Change will be included in upcoming version 0.9.0 (no exact release date known yet, sorry).

simonmaass commented:
maybe like a estimated date? weeks? months?

ngolatka commented:
Probably weeks rather than months. But in the end, it isn't our decision, but DHL's.

powli commented:
@simonmaass, @MarcusWolschon, @tuunit:

Fix released with [0.9.0](https://github.com/netresearch/dhl-module-shipping-m2/releases/tag/0.9.0).

simonmaass commented:
@powli great to hear - thank you!
